python/ray/air/result.py

```diff
@@ -106,8 +107,10 @@ def _repr(self, indent: int = 0) -> str:
             shown_attributes.pop("error")
 
         if self.metrics:
+            exclude = set(AUTO_RESULT_KEYS)
+            exclude.update(BLACKLISTED_KEYS)
             shown_attributes["metrics"] = {
-                k: v for k, v in self.metrics.items() if k not in AUTO_RESULT_KEYS
+                k: v for k, v in self.metrics.items() if k not in exclude
```

Comment:
So BLACKLISTED_KEYS will not be shown in Result and ResultGrid repr, but will still be printed by ProgressReporter on_trial_result, right? I feel like information like `time_since_restore` and `iterations_since_restore` are still important for progress tracking.

Reply:
In this PR we're just removing it from the repr.

Currently the blacklisted keys are also not printed in the results table. Per default, only the most relevant metrics should be printed. If users want to show more metrics, they can customize this.

Maybe customization of this should be a blocker for enabling per default.